src/Storages/ObjectStorageQueue/ObjectStorageQueueSource.cpp

```diff
@@ -503,6 +503,9 @@ Chunk ObjectStorageQueueSource::generateImpl()
         auto file_metadata = object_info->file_metadata;
         auto file_status = file_metadata->getFileStatus();
         const auto & path = reader.getObjectInfo()->getPath();
+        last_started_file = file_metadata;
```

Comment:
why don't we push instantly to `processed_files`?
In which case do we have it set but not present in `processed_files`?